src/FileManager.cpp

```diff
@@ -163,22 +162,45 @@ void FileManager::writeToFile(std::vector<std::string> ignore_dirs, std::vector<
     } else {
       // entry is a directory
       file << "|-- " + entry.rname + "\n";
-#ifdef DEBUG
-      std::cout << "Checking " << entry.rname << " against build" << std::endl; 
-#endif
+
+      debugln("Checking",entry.rname,"against build"); 
+
       if (itemInList(entry.rname, ignore_dirs)) {
-#ifdef DEBUG
-        std::cout << "Ignoring dir: " << entry.name << "\n";
-#endif
+
+        debugln("Ignoring dir:", entry.name);
+
         continue;
       } else {
         this->clear(entry.name);
         this->writeToFileIterated(file, depth, ignore_dirs, ignore_extensions);
       }
     }
   }
+  debugln("Done!");
```

Comment:
Are we missing `ifdef` and `#endif` here?

Reply:
No, `#ifdef` and `#endif` are being added in the variadic template itself so that the code looks cleaner here.